src/app/components/OptimizelyPageMetrics/index.jsx

```diff
@@ -1,21 +1,42 @@
 import React, { useState, useContext, useEffect, useRef } from 'react';
 import { OptimizelyContext } from '@optimizely/react-sdk';
 import { RequestContext } from '#contexts/RequestContext';
+import useOptimizelyScrollDepth from '#hooks/useOptimizelyScrollDepth';
 import useOptimizelyVariation from '#hooks/useOptimizelyVariation';
 import OPTIMIZELY_CONFIG from '#lib/config/optimizely';
 
-const OptimizelyArticleCompleteTracking = () => {
+const OptimizelyPageMetrics = ({ pageView, pageDepth, pageComplete }) => {
```

Comment:
I wonder if we give these params more descriptive names, because I'm a bit confused 😅 (especially when they are used with lots of other booleans to determine whether to send events!)

```suggestion
const OptimizelyPageMetrics = ({ trackPageView, trackPageDepth, trackPageComplete }) => {
```

Reply:
Should we give them default values (perhaps false?)

Also, I wonder if this is an opportunity for Typescript - this will massively help with knowing what the types of these variables are & improve DevX

Reply:
Yep absolutely - refactor incoming!